### PR TITLE
Support building and running with full MIR on foreign architectures, drop support for missing MIR

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes https://github.com/solson/miri/issues/211, fixes https://github.com/solson/miri/issues/202

### CODE REVIEW

**Comment by oli-obk** [comment]
Awesome! r=me with appveyor succeeding

**Comment by RalfJung** [comment]
I added a small change to also use `rustc_version` in the test suite to determine the host triple.

File: tests/compiletest.rs [hunk 1]
```diff
@@ -142,17 +142,15 @@ fn get_sysroot() -> PathBuf {
...
-    let host = std::str::from_utf8(&host).expect("sysroot is not utf8");
-    let host = host.split("\nhost: ").nth(1).expect(
-        "no host: part in rustc -vV",
-    );
-    let host = host.split('\n').next().expect("no \n after host");
-    String::from(host)
+    let rustc_version = std::str::from_utf8(&rustc_version).expect("rustc -vV is not utf8");
+    let version_meta = rustc_version::version_meta_for(&rustc_version)
+        .expect("failed to parse rustc version info");
+    version_meta.host
```
Comment (oli-obk) [inline tests/compiletest.rs]: The entire `get_host` function body could be just `rustc_version::version_meta().unwrap().host`

Comment (RalfJung) [reply tests/compiletest.rs]: No, this uses a different binary than `rustc` when being run in the rustc test suite.

Comment (oli-obk) [reply tests/compiletest.rs]: oh... because of the `RUSTC` env var. Oh well, it's not very important.

**Pull request merged**